integration-templates/calendly/tests/calendly-event-invitees.test.ts

```diff
@@ -0,0 +1,114 @@
+        nangoMock.batchDelete = vi.fn().mockReturnValue(batchDeleteData) as NangoSync['batchDelete'];
+    }
+
+    if (getConnection) {
+        nangoMock.getConnection = vi.fn().mockReturnValue(getConnection) as NangoSync['getConnection'];
+    }
+
+    if (paginateGet || paginatePost || paginatePatch || paginatePut || paginateDelete) {
+        nangoMock.paginate = vi.fn().mockImplementation(async function* (args: any) {
+            const { endpoint, method = 'GET' } = args;
+            switch (method) {
+                case 'GET': {
+                    const { default: paginateDataGet } = await import(`../calendly/mocks/paginate/get/event-invitees${endpoint}.json`, {
+                        assert: { type: 'json' }
+                    });
+                    paginateGet = paginateDataGet;
+                    yield paginateGet;
+                    break;
+                }
+                case 'POST': {
+                    const { default: paginateDataPost } = await import(`../calendly/mocks/paginate/post/event-invitees${endpoint}.json`, {
+                        assert: { type: 'json' }
+                    });
+                    paginatePost = paginateDataPost;
+                    yield paginatePost;
+                    break;
+                }
+                case 'PATCH': {
+                    const { default: paginateDataPatch } = await import(`../calendly/mocks/paginate/patch/event-invitees${endpoint}.json`, {
+                        assert: { type: 'json' }
+                    });
+                    paginatePatch = paginateDataPatch;
+                    yield paginatePatch;
+                    break;
+                }
+                case 'PUT': {
+                    const { default: paginateDataPut } = await import(`../calendly/mocks/paginate/put/event-invitees${endpoint}.json`, {
+                        assert: { type: 'json' }
+                    });
+                    paginatePut = paginateDataPut;
+                    yield paginatePut;
+                    break;
+                }
+                case 'DELETE': {
+                    const { default: paginateDataDelete } = await import(`../calendly/mocks/paginate/delete/event-invitees${endpoint}.json`, {
+                        assert: { type: 'json' }
+                    });
+                    paginateDelete = paginateDataDelete;
+                    yield paginateDelete;
+                    break;
+                }
+            }
+        }) as NangoSync['paginate'];
+    }
+
+    it('should get, map correctly the data and batchSave the result', async () => {
+        await fetchData(nangoMock);
+
+        // eslint-disable-next-line @typescript-eslint/unbound-method
+        expect(nangoMock.batchSave).toHaveBeenCalledWith(batchSaveData, 'EventInvitee');
+    });
+
+    if (batchDeleteData && batchDeleteData.length > 0) {
+        it('should get, map correctly the data and batchDelete the result', async () => {
```

Comment:
There is a built-in feature to do this and also track when not runned

```suggestion
        it.runIf(batchDeleteData && batchDeleteData.length > 0)('should get, map correctly the data and batchDelete the result', async () => {
```

Reply:
This is throwing an error, can you point to the docs where this feature is described? All I see is something like this https://stackoverflow.com/questions/58264344/conditionally-run-tests-in-jest

Reply:
https://vitest.dev/api/#test-runif

Reply:
Thanks, updated!